metrics/segmentStats.py

```diff
@@ -0,0 +1,372 @@
+                    simplePaths = list(nx.all_simple_paths(self._subGraphSkeleton, source=sourceOnTree, target=item))
+                    simplePath = simplePaths[0]
+                    countBranchNodesOnPath = sum([1 for point in simplePath if point in nodes])
+                    if countBranchNodesOnPath == 2:
+                        curveLength = self._getLengthAndRemoveTracedPath(simplePath)
+                        self.isolatedEdgeInfoDict[sourceOnTree, item] = curveLength
+        else:
+            # each node is connected to one or two other nodes implies it is a line,
+            endPoints = [k for (k, v) in self._nodeDegreeDict.items() if v == 1]
+            sourceOnLine = endPoints[0]
+            targetOnLine = endPoints[1]
+            simplePath = nx.shortest_path(self._subGraphSkeleton, source=sourceOnLine, target=targetOnLine)
+            curveLength = self._getLengthAndRemoveTracedPath(simplePath)
+            self.isolatedEdgeInfoDict[sourceOnLine, targetOnLine] = curveLength
+
+    def _setCountDict(self, source):
+        if source not in self._visitedSources:
+            self.countDict[source] = 1
+            self._visitedSources.append(source)
+        else:
+            self.countDict[source] = self.countDict[source] + 1
+
+    def _setHausdorffDimensionDict(self, source, target, length, displacement):
+        logDisplacement = np.log(displacement)
+        if logDisplacement != -np.inf or not np.allclose(logDisplacement, 0.0):
+            self.hausdorffDimensionDict[self.countDict[source], source, target] = np.log(length) / logDisplacement
+
+    def _singleCycle(self, cycle):
+        """
+        Find statistics of a single cycle of a disjoint graph
+        """
+        sourceOnCycle = cycle[0]
+        self._setCountDict(sourceOnCycle)
+        curveLength = self._getLengthAndRemoveTracedPath(cycle, isCycle=True)
+        self.cycleInfoDict[self.cycles] = [0, curveLength]
+        self.lengthDict[self.countDict[sourceOnCycle], sourceOnCycle, cycle[len(cycle) - 1]] = curveLength
+        self.tortuosityDict[self.countDict[sourceOnCycle], sourceOnCycle, cycle[len(cycle) - 1]] = 0
+        self.contractionDict[self.countDict[sourceOnCycle], sourceOnCycle, cycle[len(cycle) - 1]] = 0
+        self.cycles = self.cycles + 1
+
+    def _cyclicTree(self, cycleList):
+        """
+        Find statistics of a cyclic tree of a disjoint graph
+        """
+        for nthcycle, cycle in enumerate(cycleList):
+            nodeDegreeDictFilt = {key: value for key, value in self._nodeDegreeDict.items() if key in cycle}
+            branchPointsOnCycle = [k for (k, v) in nodeDegreeDictFilt.items() if v != 2 and v != 1]
+            sourceOnCycle = branchPointsOnCycle[0]
+            if len(branchPointsOnCycle) == 1:
+                self._singleCycle(cycle)
+            else:
+                for point in cycle:
+                    if point not in branchPointsOnCycle:
+                        continue
```

Comment:
For better code clarity don't use continue. Instead restructure your if statements to achieve the desired for flow control.